File: ProjetoCs-Reserva_de_Hotel-main/ReservaDeHotel/Models/pagamento.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace ReservaDeHotel.Models;
public class Pagamento
{
    [Key]
    public int IdPagamento {get;set;}
    public string? MetodoPagamento { get; set;}
    public decimal Valor { get; set;} 
    public DateTime DataPagamento { get; set;}

    [ForeignKey("ReservaHotel")]
    public int IdReservaEstadia { get; set;} // aqui teria que usar uma ForeignKey
    public ReservaHotel ReservaHotel { get; set;}
}

File: ReservaHotelFront/src/app/Pagamento.ts
import { ReservaHotel } from './ReservaHotel';

export class Pagamento {
    IdPagamento: number = 0;
    MetodoPagamento: string | null = null;
    Valor: number = 0;
    DataPagamento: Date = new Date();
    IdReservaEstadia: number = 0;
    ReservaHotel: ReservaHotel | null = null;
}
